app/assets/javascripts/controllers/bootstrap_tree_controller.js

```diff
@@ -0,0 +1,58 @@
+ManageIQ.angular.app.controller('bootstrapTreeController', ['$http', '$scope', 'bootstrapTreeSubscriptionService', 'initialTreeData', 'railsControllerName', function($http, $scope, bootstrapTreeSubscriptionService, initialTreeData, railsControllerName) {
+  var init = function() {
+    updateTree(initialTreeData);
+    $scope.rootNode = $('#bootstrap-tree').treeview('getNodes')[0];
+    $('#bootstrap-tree').treeview('selectNode', $scope.rootNode);
+
+    bootstrapTreeSubscriptionService.subscribeToTreeUpdates(updateTree);
+    bootstrapTreeSubscriptionService.subscribeToCancelClicks(selectRootNode);
+    bootstrapTreeSubscriptionService.subscribeToDeselectTreeNodes(unselectAllNodes);
+    bootstrapTreeSubscriptionService.subscribeToSingleItemSelected(selectSingleNode);
+  };
+
+  var updateTree = function(data) {
+    $('#bootstrap-tree').treeview({
+      collapseIcon: 'fa fa-angle-down',
+      data: data,
+      expandIcon: 'fa fa-angle-right',
+      nodeIcon: 'fa fa-folder',
+      showBorder: false,
+      onNodeSelected: nodeSelectedCallback
+    });
+  };
+
+  var nodeSelectedCallback = function(_event, data) {
```

Comment:
Oh, you mean because you're just calling `$http` inside. Yes, you're right, it's not strictly needed given that the function does just that. The thing is, when you're adding something else to the function in a few months, you're not gonna remember that this one is magic and runs outside of angular's digest cycle. You'll just add it and spend hours trying to figure out why it only works sometimes :). So that's why.